cli/build.go

```diff
@@ -0,0 +1,214 @@
+	oci              bool
+}
+
+func newBuildCmd(out io.Writer) *cobra.Command {
+	r := &buildCmd{
+		out: out,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "build [OPTIONS] PATH | URL",
+		Short: "Run a build",
+		Long:  buildLongDesc,
+		RunE:  r.run,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				return errors.New("acb build requires exactly 1 argument. See acb build --help")
+			}
+
+			return nil
+		},
+	}
+
+	f := cmd.Flags()
+
+	// Build parameters
+	f.StringVarP(&r.dockerfile, "file", "f", "Dockerfile", "name of the Dockerfile")
+	f.StringArrayVarP(&r.tags, "tag", "t", []string{}, "name and optionally a tag in the 'name:tag' format")
+	f.StringArrayVar(&r.buildArgs, "build-arg", []string{}, "set build time arguments")
+	f.StringArrayVar(&r.secretBuildArgs, "secret-build-arg", []string{}, "set secret build arguments")
+
+	f.StringVarP(&r.registry, "registry", "r", "", "the name of the registry")
+	f.StringVarP(&r.registryUserName, "username", "u", "", "the username to use when logging into the registry")
+	f.StringVarP(&r.registryPassword, "password", "p", "", "the password to use when logging into the registry")
+
+	f.StringVar(&r.isolation, "isolation", "default", "the isolation to use")
+	f.StringVar(&r.target, "target", "", "specify a stage to build")
+	f.BoolVar(&r.pull, "pull", false, "attempt to pull a newer version of the base images")
+	f.BoolVar(&r.noCache, "no-cache", false, "true to ignore all cached layers when building the image")
+	f.BoolVar(&r.push, "push", false, "push on success")
+	f.BoolVar(&r.oci, "oci", false, "use the OCI builder")
+
+	if debug {
+		logrus.SetLevel(logrus.DebugLevel)
+	}
+
+	return cmd
+}
+
+func (b *buildCmd) run(cmd *cobra.Command, args []string) error {
+	if err := b.validateCmdArgs(); err != nil {
+		return err
+	}
+
+	b.context = args[0]
+
+	normalizedDockerImages := builder.GetNormalizedDockerImageNames(b.tags)
+
+	cmder := cmder.NewCmder(false)
+
+	defaultStep := &graph.Step{
+		UseLocalContext: true,
+		Run:             b.createRunCmd(),
+	}
+
+	steps := []*graph.Step{defaultStep}
+
+	push := []string{}
+	if b.push {
+		for _, img := range normalizedDockerImages {
+			push = append(push, img)
+		}
+	}
+
+	// TODO: create secrets
+	secrets := []*graph.Secret{}
+
+	p := graph.NewPipeline(steps, push, secrets)
+	dag, err := graph.NewDagFromPipeline(p)
+	if err != nil {
+		return err
+	}
+
+	timeout := time.Duration(p.TotalTimeout) * time.Second
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	homeVolName := fmt.Sprintf("%s%s", volume.VolumePrefix, uuid.New())
+	fmt.Printf("Setting up the home volume: %s...\n", homeVolName)
+	v := volume.NewVolume(homeVolName, cmder)
+	if err := v.Create(ctx); err != nil {
+		return fmt.Errorf("Err creating docker vol: %v", err)
+	}
+	defer func() {
+		if err := v.Delete(ctx); err != nil {
+			fmt.Printf("Failed to clean up docker vol: %s. Err: %v\n", homeVolName, err)
```

Comment:
>fmt.Printf [](start = 3, length = 10)

I notice the new code most time just use Printf for logging rather than logrus. Is it intentional or temporary?

Reply:
Temporary, it was always a grab-bag of logrus before; I'm not particular for one or the other. I don't see clear value in having logrus other than color right now. Made a work item to refactor it all with a logging abstraction.